File: Server/Server/Models/Blogs/Blog.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Server.Models.Auth;

namespace Server.Models.Blogs
{
    public class Blog
    {
        [Key]
        public Guid BlogId { get; set; } = Guid.NewGuid();

        [Required]
        public Guid AuthorUserId { get; set; }

        [ForeignKey("AuthorUserId")]
        public virtual User? Author { get; set; }

        [Required]
        [MaxLength(300)]
        public string Title { get; set; } = string.Empty;

        [Required]
        public string Content { get; set; } = string.Empty; // CLOB

        public bool IsPublished { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.Now;

        public DateTime? UpdatedAt { get; set; }

        public virtual ICollection<BlogImage> Images { get; set; } = new List<BlogImage>();
    }
}

File: Client/app/types/blog.ts
export interface BlogImage {
    blogImageId: string;
    blogId: string;
    filePath: string;
    caption?: string;
    isPrimary: boolean;
    sortOrder: number;
}

export interface Blog {
    blogId: string;
    authorUserId: string;
    authorName: string;
    authorAvatar: string;
    title: string;
    content: string;
    isPublished: boolean;
    createdAt: string;
    updatedAt?: string;
    images: BlogImage[];
}

export interface CreateBlogDto {
    title: string;
    content: string;
    isPublished?: boolean;
}

export interface UpdateBlogDto {
    title: string;
    content: string;
    isPublished: boolean;
}

export interface BlogImageCreateDto {
    filePath: string;
    fileName: string;
    fileType: string;
    fileSize: number;
    caption?: string;
    isPrimary?: boolean;
}
